pkg/aws/ec2/pricing_map.go

```diff
@@ -65,21 +66,42 @@ type Prices struct {
 	Total float64
 }
 
-func NewStructuredPricingMap() *StructuredPricingMap {
-	return &StructuredPricingMap{
+// StorageStructuredPricingMap collects a map of StoragePricing structs where the key is the region
+type StorageStructuredPricingMap struct {
+	// Regions is a map of region code to StoragePricing
+	// key is the region
+	// value is a map of storage classes to prices
+	Regions map[string]*StoragePricing
+	m       sync.RWMutex
+}
+
+// StoragePricing is a map where the key is the storage type and the value is the price
+type StoragePricing struct {
+	Storage map[string]float64
+}
+
+func NewComputeStructuredPricingMap() *ComputeStructuredPricingMap {
+	return &ComputeStructuredPricingMap{
 		Regions:         make(map[string]*FamilyPricing),
-		InstanceDetails: make(map[string]Attributes),
+		InstanceDetails: make(map[string]InstanceAttributes),
 		m:               sync.RWMutex{},
 	}
 }
 
-// GeneratePricingMap accepts a list of ondemand prices and a list of spot prices.
+func NewStorageStructuredPricingMap() *StorageStructuredPricingMap {
+	return &StorageStructuredPricingMap{
+		Regions: make(map[string]*StoragePricing),
+		m:       sync.RWMutex{},
+	}
+}
+
+// GenerateComputePricingMap accepts a list of ondemand prices and a list of spot prices.
 // The method needs to
 // 1. Parse out the ondemand prices and generate a productTerm map for each instance type
 // 2. Parse out spot prices and use the productTerm map to generate a spot price map
-func (spm *StructuredPricingMap) GeneratePricingMap(ondemandPrices []string, spotPrices []ec2Types.SpotPrice) error {
+func (spm *ComputeStructuredPricingMap) GenerateComputePricingMap(ondemandPrices []string, spotPrices []ec2Types.SpotPrice) error {
```

Comment:
My IDE is yelling at me that the receivers(`spm`) for `*ComputeStructuredPricingMap` do not match. Can you update the recievers for the newly named `ComputeStructurePricingMap` to be `cspm`? Looks like it was updated in some but not all methods

Reply:
sure, but tell me how you got your IDE to yell at you for that, mine is 🤐

Reply:
I'm using GoLand and I don't think I did anything special. When I'm looking at the code there's the squiggly line to inform me of a warning and when I hover over, the IDE provides this affordance: 
![image](https://github.com/user-attachments/assets/3a4dfe05-9171-4ee3-87c1-8a0e67981812)


You can also open up the problems tab and see the warnings/errors for the file: 
![image](https://github.com/user-attachments/assets/c09134d4-6cde-4080-ba37-8e5d8e6737ee)

Reply:
ah, yeah, JetBrains will do that for you ❤️  I'm on VSCode lately since I wasn't doing that much Go at all. Might need to find a plugin or reconsider my choices in life...

Reply:
since I'm at it, what do you think about dropping the `structured` bit of this naming altogheter? I find that it makes for a long name and doesn't really add much information. It's a struct. We get that it'll be structured, no?

Reply:
I'm in favor of renaming. It made sense early on in the project(S3) and I don't think it does anymore. I'm fine to making it just `ComputePricingMap` and `StoragePricingMap` 👍🏻

Reply:
done